src/pathfinding_service/service.py

```diff
@@ -114,162 +106,152 @@ def __init__(
             log.critical('Could not connect to broadcasting system.', exc=e)
             sys.exit(1)
 
-    def _setup_token_networks(self) -> None:
-        self.token_network_registry_listener.add_confirmed_listener(
-            create_registry_event_topics(CONTRACT_MANAGER), self.handle_token_network_created
-        )
-
     def _run(self) -> None:  # pylint: disable=method-hidden
         register_error_handler(error_handler)
         self.matrix_listener.start()
-        self.token_network_registry_listener.start()
-        self.is_running.wait()
+        while not self.is_running.is_set():
+            last_confirmed_block = self.web3.eth.blockNumber - self.required_confirmations
+
+            last_query_interval_block = (
```

Comment:
I had problems with this name because I read it as "last (query_interval_block)" and didn't know what a query interval block should be. Might just be me, though.